src/sdk/cloudformation.ts

```diff
@@ -0,0 +1,112 @@
+import AWS, { CloudFormation } from 'aws-sdk'
+
+import config from 'config'
+import { MiraObject } from '../cdk/object'
+import { getRoleArn, assumeRole } from '../assume-role'
+import { MiraConfig } from '../config/mira-config'
+
+let miraCfn: AWS.CloudFormation
+/**
+ * Gets a Cfn object.
+ */
+export const getCfn = async (): Promise<CloudFormation> => {
+  if (miraCfn) {
+    return miraCfn
+  }
+  const awsConfig = await assumeRole(getRoleArn(config.get(`accounts.${MiraConfig.getEnvironment().name}.profile`)))
+  awsConfig.region = MiraConfig.getEnvironment().env.region
+  AWS.config = awsConfig
+  miraCfn = new AWS.CloudFormation({ apiVersion: '2010-05-15' })
+  return miraCfn
+}
+
+/**
+ * Gets the stacks that are in progress.  This can be used to detect if a stack
+ * is in progress of being pushed.
+ * @todo
+ */
+// export const getInProgressStacks = (): void => {
+
+// }
+
+/**
+ * Gets the resources included in a stack.
+ */
+export const getStackResources = async (StackName: string): Promise<LooseObject> => {
+  const cfn = await getCfn()
+  const result = await cfn.listStackResources({
+    StackName
+  }).promise()
+  return result as LooseObject
+}
+
+/**
+ * Gets all stacks that have a state suffix of `_COMPLETE`.
+ */
+export const getStacks = async (filter?: string): Promise<LooseObject> => {
+  const cfn = (await getCfn())
+  const result = await cfn.listStacks({
+    StackStatusFilter: [
+      'IMPORT_COMPLETE',
+      'UPDATE_ROLLBACK_COMPLETE',
+      'UPDATE_COMPLETE',
+      'ROLLBACK_COMPLETE',
+      'CREATE_COMPLETE'
+    ]
+  }).promise()
+  if (result && result.StackSummaries) {
+    let stacks = result.StackSummaries
+    if (filter) {
+      stacks = result.StackSummaries.filter(({ StackName }) => StackName.startsWith(filter))
+    }
+    return stacks as LooseObject
+  }
+  return {} as LooseObject
+}
+
+/**
+ * Gets all stacks are associated with this Mira instance.
+ */
+export const getInstaceStacks = async (): Promise<LooseObject> => {
+  const stackName = (MiraObject.calculateSharedResourceName('stack'))
```

Comment:
Why isn't `MiraConfig` used here instead of `MireObject`? This is the only static method on `MiraObject` that calls to `MiraConfig` without changing anything on the input/output. Seems like very specific to this use case.

Reply:
I think the goal is to wind up segmenting out the usages of the Config so that the problems the Config class solves is more discrete.  Where we are wrapping CDK consturcts, we should use MiraObject.  Where we parse Config files, we should use MiraConfig.